### PR TITLE
Add a new code action to find cases where an expression could be updated to use a collection initializer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide    VB impl and tests coming shortly.

**Comment by CyrusNajmabadi** [comment]
test this please

File: src/EditorFeatures/CSharpTest/UseCollectionInitializer/UseCollectionInitializerTests.cs [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+        {
+            await TestMissingAsync(
+@"
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        var c = [||]new C();
+        c.Add(1);
```
Comment (jmarolf) [inline src/EditorFeatures/CSharpTest/UseCollectionInitializer/UseCollectionInitializerTests.cs]: >c.Add(1); [](start = 8, length = 9)

What if C actually has an Add method?

File: src/EditorFeatures/VisualBasicTest/UseCollectionInitializer/UseCollectionInitializerTests.vb [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+Class C
+    Sub M()
+        Dim c = [||]New List(Of Integer)()
+        c.Add
+    End Sub
+End Class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCollectionInitializer)>
+        Public Async Function TestMissingOnOmittedArg() As Task
```
Comment (jmarolf) [inline src/EditorFeatures/VisualBasicTest/UseCollectionInitializer/UseCollectionInitializerTests.vb]: >TestMissingOnOmittedArg [](start = 30, length = 23)

Is there a test for this case in C#?

**Comment by jmarolf** [comment]
:shipit:

File: src/EditorFeatures/CSharpTest/UseCollectionInitializer/UseCollectionInitializerTests.cs [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+            1, // Foo
+            2 // Bar
+        };
+    }
+}",
+compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCollectionInitializer)]
+        public async Task TestComplexInitializer2()
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/UseCollectionInitializer/UseCollectionInitializerTests.cs]: Great idea.  I'm going to add that in a followup PR.

**Pull request merged**